cmd/juju/secretbackends/modelsecretbackend_test.go

```diff
@@ -0,0 +1,115 @@
+	return ctrl
+}
+
+func (s *modelSecretBackendCommandSuite) TestGet(c *gc.C) {
+	defer s.setup(c).Finish()
+
+	s.secretsAPI.EXPECT().GetModelSecretBackend(gomock.Any()).Return("myVault", nil)
+	s.secretsAPI.EXPECT().Close().Return(nil)
+
+	ctx, err := cmdtesting.RunCommand(c, secretbackends.NewModelCredentialCommandForTest(s.store, s.secretsAPI))
+	c.Assert(err, jc.ErrorIsNil)
+	out := cmdtesting.Stdout(ctx)
+	c.Assert(out, gc.Equals, "myVault"+"\n")
+}
+
+func (s *modelSecretBackendCommandSuite) TestGetNotSupported(c *gc.C) {
+	defer s.setup(c).Finish()
+
+	s.secretsAPI.EXPECT().GetModelSecretBackend(gomock.Any()).Return("", errors.NotSupportedf("model secret backend"))
+	s.secretsAPI.EXPECT().Close().Return(nil)
+
+	_, err := cmdtesting.RunCommand(c, secretbackends.NewModelCredentialCommandForTest(s.store, s.secretsAPI))
+	c.Assert(err, gc.ErrorMatches, `"model-secret-backend" has not been implemented on the controller, use the "model-config" command instead`)
+}
+
+func (s *modelSecretBackendCommandSuite) TestSet(c *gc.C) {
+	defer s.setup(c).Finish()
+
+	s.secretsAPI.EXPECT().SetModelSecretBackend(gomock.Any(), "myVault").Return(nil)
+	s.secretsAPI.EXPECT().Close().Return(nil)
+
+	_, err := cmdtesting.RunCommand(c, secretbackends.NewModelCredentialCommandForTest(s.store, s.secretsAPI), "myVault")
+	c.Assert(err, jc.ErrorIsNil)
+}
+
+func (s *modelSecretBackendCommandSuite) TestSetNotSupported(c *gc.C) {
+	defer s.setup(c).Finish()
+
+	s.secretsAPI.EXPECT().SetModelSecretBackend(gomock.Any(), "myVault").Return(errors.NotSupportedf("model secret backend"))
+	s.secretsAPI.EXPECT().Close().Return(nil)
+
+	_, err := cmdtesting.RunCommand(c, secretbackends.NewModelCredentialCommandForTest(s.store, s.secretsAPI), "myVault")
+	c.Assert(err, gc.ErrorMatches, `"model-secret-backend" has not been implemented on the controller, use the "model-config" command instead`)
+}
+
+func (s *modelSecretBackendCommandSuite) TestSetSecretBackendNotFound(c *gc.C) {
+	defer s.setup(c).Finish()
+
+	s.secretsAPI.EXPECT().SetModelSecretBackend(gomock.Any(), "myVault").Return(secretbackenderrors.NotFound)
+	s.secretsAPI.EXPECT().Close().Return(nil)
+
+	_, err := cmdtesting.RunCommand(c, secretbackends.NewModelCredentialCommandForTest(s.store, s.secretsAPI), "myVault")
+	c.Assert(err, gc.ErrorMatches, `secret backend not found: please use "add-secret-backend" to add "myVault" to the controller first`)
+}
+
+func (s *modelSecretBackendCommandSuite) TestSetSecretBackendNotValid(c *gc.C) {
+	defer s.setup(c).Finish()
+
+	s.secretsAPI.EXPECT().SetModelSecretBackend(gomock.Any(), "internal").Return(secretbackenderrors.NotValid)
+	s.secretsAPI.EXPECT().Close().Return(nil)
+
+	_, err := cmdtesting.RunCommand(c, secretbackends.NewModelCredentialCommandForTest(s.store, s.secretsAPI), "internal")
+	c.Assert(err, gc.ErrorMatches, `secret backend not valid: please use "auto" instead`)
+}
+
+func (s *modelSecretBackendCommandSuite) TestSetFailedEmptyBackendName(c *gc.C) {
```

Comment:
I can't see how this case is covered or produced. I am lost at this test and how this contract is satisfied. I am missing something but can't see what.